win32/glob.c

```diff
@@ -443,9 +535,23 @@ static int glob0( const Char *pattern, glob_t *pglob)
 				*bufnext++ = M_NOT;
 			c = *qpatnext++;
 			do {
+				if (c == LBRACKET && *qpatnext == ':') {
+					do {
+						err = g_charclass(&qpatnext,
+						    &bufnext);
+						if (err)
+							break;
+						c = *qpatnext++;
+					} while (c == LBRACKET && *qpatnext == ':');
+					if (err == -1 &&
+					    !(pglob->gl_flags & GLOB_NOCHECK))
+						return GLOB_NOMATCH;
+					if (c == RBRACKET)
+						break;
+				}
 				*bufnext++ = CHAR(c);
 				if (*qpatnext == RANGE &&
-					(c = qpatnext[1]) != RBRACKET) {
+				    (c = qpatnext[1]) != RBRACKET) {
```

Comment:
Indent changed from tab to spaces

Reply:
Ditto, this how OpenBSD indented it (same for the ones you mentioned below). I can change this to match PHP style.

Reply:
I prefer to keep this file as close to the original as possible, to make backports from upstream easier.

Reply:
I don't mind doing the indentation changes if that's how we do it; the file doesn't change too much and in the future we can probably look at individual patches. I just did a blanket import since it looks like we had almost 20 years of changes to deal with.

Reply:
> I prefer to keep this file as close to the original as possible, to make backports from upstream easier.

Well, I suppose if we can backport the indent issues I'm fine with that too

Reply:
I wouldn't worry that much about diverging from upstream as we might want to do some changes into it eventually. It would be however useful to track the version that it is based on so we can from time to time review changes in the upstream and backport it to our implementation.